bot/cogs/reddit.py

```diff
@@ -19,8 +21,13 @@
 class Reddit(Cog):
     """Track subreddit posts and show detailed statistics about them."""
 
-    HEADERS = {"User-Agent": "Discord Bot: PythonDiscord (https://pythondiscord.com/)"}
+    # Change your client's User-Agent string to something unique and descriptive,
+    # including the target platform, a unique application identifier, a version string,
+    # and your username as contact information, in the following format:
+    # <platform>:<app ID>:<version string> (by /u/<reddit username>)
+    USER_AGENT = "docker:Discord Bot of https://pythondiscord.com/:v?.?.? (by /u/PythonDiscord)"
```

Comment:
`platform` should probably be something like `python3`, I don't think `docker` makes a lot of sense here.

Also, the versioning doesn't have lots of value. I think we should export an environment variable like `BUILD_COMMIT_SHA` in the build process via `git` and use that here instead.

Reply:
Reddits API rules used `android` as an example for platform, so I deemed `docker` to be fitting. We could also be more descriptive and change it to something like `docker-python` or maybe even `docker-python/aiohttp`.

Using `BUILD_COMMIT_SHA` as version string sounds good to me.

Reply:
You can also do:
```py
sha = subprocess.run(['git', 'rev-parse', '--short', 'HEAD'], stdout=subprocess.PIPE).stdout.decode('utf-8')
```
to get the current commit sha1.

Reply:
I've found, Azure Pipelines has a `BuildNumber` variable, which would be more fitting to use as a version value, imho

However, I haven't used Azure Pipelines and Docker prior to this bot, so implementing it is a bit tricky. ^^

Reply:
I really think it's not worth the hassle. The version number in the user agent is ultimately quite inconsequential. We can just hard-code 1.0.0 or something. Thoughts?